pkg/kev/config/extensions.go

```diff
@@ -0,0 +1,143 @@
+package config
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/imdario/mergo"
+	"gopkg.in/yaml.v3"
+)
+
+type ExtensionRoot struct {
+	K8S K8SServiceConfig `yaml:"x-k8s"`
+}
+
+type K8SServiceConfig struct {
+	Enabled   bool      `yaml:"enabled"`
+	Workload  Workload  `yaml:"workload" validate:"required"`
+	Component Component `yaml:"component"`
+}
+
+func DefaultK8SServiceConfig() K8SServiceConfig {
+	return K8SServiceConfig{
+		Enabled: DefaultServiceEnabled,
+		Workload: Workload{
+			Type:           DefaultWorkload,
+			LivenessProbe:  DefaultLivenessProbe(),
+			ReadinessProbe: DefaultReadinessProbe(),
+		},
+	}
+}
+
+func K8SServiceCfgFromMap(m map[string]interface{}) (K8SServiceConfig, error) {
+	var extensions ExtensionRoot
+
+	var buf bytes.Buffer
+	if err := yaml.NewEncoder(&buf).Encode(m); err != nil {
+		return K8SServiceConfig{}, err
+	}
+
+	if err := yaml.NewDecoder(&buf).Decode(&extensions); err != nil {
+		return K8SServiceConfig{}, err
+	}
+
+	err := validator.New().Struct(extensions.K8S.Workload)
+	if err != nil {
+		validationErrors := err.(validator.ValidationErrors)
+		for _, e := range validationErrors {
+			if e.Tag() == "required" {
+				return K8SServiceConfig{}, fmt.Errorf("%s is required", e.StructNamespace())
+			}
+		}
+
+		return K8SServiceConfig{}, errors.New(validationErrors[0].Error())
+	}
+
+	k8s := DefaultK8SServiceConfig()
+
+	if err := mergo.Merge(&extensions.K8S, k8s); err != nil {
+		return K8SServiceConfig{}, err
+	}
+
+	return extensions.K8S, nil
+}
+
+// TODO: hyphenated words suck, should we do camelCase?
```

Comment:
Yes, I think it makes sense to switch to camelCase in extensions. Ideally let's do it as part of this initial PR please.